### PR TITLE
BUG: Last name should be input when singing up

### PR DESCRIPTION
**Pull request by darkheros12**
I added a last name field.

### CODE REVIEW

File: app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java [hunk 1]
```diff
@@ -19,26 +19,17 @@ public SignUpViewModel() {
...
     }
 
-    public String nameParcer(String firstName, String lastName) {
-        return firstName + " " + lastName;
-    }
-
-    public String[] nameExtractor(String name) {
-        return name.split(" ", 2);
-    }
-
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java:41]: Nice. This was kind of a hack, so glad this is gone now that there are 2 inputs

File: app/src/main/java/com/example/bigowlapp/activity/SignUpPageActivity.java [hunk 2]
```diff
@@ -40,12 +41,17 @@ protected void initialize() {
...
 
             //Error handling
-            if (userName.isEmpty()) {
-                username.setError("Please enter your name");
-                username.requestFocus();
+            if (firstName.isEmpty()) {
+                userFirstName.setError("Please enter your first name");
+                userFirstName.requestFocus();
+            } else if (lastName.isEmpty()) {
+                userLastName.setError("Please enter your first name");
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/activity/SignUpPageActivity.java:53]: Lol. Copy paste error. You mean:
 `userLastName.setError("Please enter your last name");`
wrong string

File: app/src/main/java/com/example/bigowlapp/repository/AuthRepository.java [hunk 3]
```diff
@@ -68,8 +68,8 @@ public FirebaseUser getCurrentUser() {
         });
         return taskBoolean;
     }
-    
-    public void addAuthStateListener(FirebaseAuth.AuthStateListener authStateListener){
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/repository/AuthRepository.java:72]: good catch

File: app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java [hunk 4]
```diff
@@ -19,26 +19,17 @@ public SignUpViewModel() {
...
+                    group.setName(firstName + "'s group " + "#" + randomStringGenerator());
                     groupRepository.addDocument(group);
                 });
 
     }
 
-    public String nameParcer(String firstName, String lastName) {
-        return firstName + " " + lastName;
-    }
-
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java:37]: For this, you should add back that nameParcer fuction (although rename it like `getFullName()`), and set group name using that.

We should keep the original behavior of that code in line 27/28 where the groups name used the full name

File: app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java [hunk 5]
```diff
@@ -19,26 +19,17 @@ public SignUpViewModel() {
     }
 
     // TODO: Create a default group upon user creation
```
Comment (refatK) [inline app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java:21]: oops, you can remove this TODO. Left this by accident, its completed

**Review by refatK — CHANGES_REQUESTED** [review]
Just some small changes, but looks good!

File: app/src/main/java/com/example/bigowlapp/activity/SignUpPageActivity.java (hunk 2 above)
Comment (darkheros12) [reply app/src/main/java/com/example/bigowlapp/activity/SignUpPageActivity.java:53]: thats true....

File: app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java (hunk 5 above)
Comment (darkheros12) [reply app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java]: I removed it.

File: app/src/main/java/com/example/bigowlapp/activity/SignUpPageActivity.java (hunk 2 above)
Comment (darkheros12) [reply app/src/main/java/com/example/bigowlapp/activity/SignUpPageActivity.java]: I changed the string

File: .idea/gradle.xml [hunk 6]
```diff
@@ -15,6 +15,7 @@
           </set>
         </option>
         <option name="resolveModulePerSourceSet" value="false" />
+        <option name="useQualifiedModuleNames" value="true" />
       </GradleProjectSettings>
     </option>
   </component>
```
Comment (refatK) [inline .idea/gradle.xml:21]: Please stop pushing things from the auto-generated idea folder. It's unnecessary

**Review by refatK — APPROVED** [review]
Thanks for addressing my comments.

Someone else should run the app and check it works as expected.

File: .idea/gradle.xml (hunk 6 above)
Comment (darkheros12) [reply .idea/gradle.xml:21]: I dont know why it keep pushing. I unchecked it so many times.

File: app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java [hunk 7]
```diff
@@ -18,27 +18,21 @@ public SignUpViewModel() {
         groupRepository = new GroupRepository();
     }
 
-    // TODO: Create a default group upon user creation
-    public Task<Boolean> createUser(String email, String password, String phoneNumber, String name) {
-        return authRepository.signUpUser(email, password, phoneNumber, nameExtractor(name)[0],
-                nameExtractor(name)[1])
+    public Task<Boolean> createUser(String email, String password, String phoneNumber, String firstName, String lastName) {
```
Comment (KhangZhengNg) [inline app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java:21]: Nice change here in the create method after you changed the sign up page

File: app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java [hunk 8]
```diff
@@ -18,27 +18,21 @@ public SignUpViewModel() {
...
 
-    public String nameParcer(String firstName, String lastName) {
+    public String getFullName(String firstName, String lastName) {
         return firstName + " " + lastName;
     }
 
-    public String[] nameExtractor(String name) {
-        return name.split(" ", 2);
-    }
-
```
Comment (KhangZhengNg) [inline app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java:41]: Well done on refactor and remove unsed code

**Review by KhangZhengNg — APPROVED** [review]
Tested and worked on both my phone and virtual device. well done

**Pull request merged**